### ISSUE TITLE
Unable to customize or replace ServoMetricCollector

### ISSUE THREAD

**Issue by brenuart** [spring-cloud/spring-cloud-netflix#346]
`ServoMetricCollector` is a concrete class whose purpose is to periodically synchronise servo metrics with Spring Boot's metrics framework. 

When auto configuration is enabled, an instance of that class is created by `org.springframework.cloud.netflix.servo.ServoMetricsAutoConfiguration` at the condition that no instance of the same type already exists in the application context. So if a different synchronisation behaviour is needed, one should create himself an instance of `ServoMetricCollector`.

Unfortunately, `ServoMetricCollector` is hardly customisable:
- the initialisation of the polling mechanism (frequency and actual poller implementation) is performed in the constructor
- the `MetricObserver` implementation that performs the actual work is implemented as a `private static final` inner class

In short, default behaviour proposed by the auto configuration can only be bypassed if one registers a custom instance of `ServoMetricCollector` with some part overridden - but the class itself doesn't allow for any useful subclassing... Stuck.

Different approaches are possible:
- refactor the `ServoMetricCollector` such that useful subclassing can be done
- change the `@Condition` in `ServoMetricAutoConfiguration` to test against a _bean name_ in the context 
- extract an interface out of the current `ServoMetricCollector`, provide a default implementation and change the `@Condition` on the factory method to test against the presence of the interface instead of the concrete class

What do you think?

**Comment by spencergibb** [comment]
I like option 1 or 3 personally.  Pull requests welcome.

**Comment by brenuart** [comment]
Servo seems to be deprecated now in SpringCloud - I think this issue can be closed now...
